### DISCUSSION TITLE
IPv6 support

### DISCUSSION THREAD

**Discussion by unclerico85** [ParallelSSH/parallel-ssh#289] [Q&A] [answered]
Hi,
Are IPv6 addresses supported by parallel-ssh? My initial attempt throws an exception. Following the traceback it would appear that only IPv4 is supported as the AF_INET family is used when creating the socket in clients/base/single.py.

Thanks

**Comment by pkittenis** [comment, marked as answer]
Hi,

Thanks for the interest. Probably not, it is untested.

Should be easy enough to support though, needs tests to ensure behaviour is correct, ie connect when ipv6 is available, use `getaddrinfo` before connecting to get available IPs etc.

**Reply by unclerico85** [reply]
Thanks for the quick response. Is this a feature that could be added to the wish list? Adding the support is a bit out of my realm of knowledge, but I could help test.

**Reply by pkittenis** [reply]
Yes, have raised a feature request for it - #289. Thanks for pointing it out.
